hummingbot/connector/exchange/binance/binance_api_order_book_data_source.py

```diff
@@ -201,30 +216,27 @@ async def listen_for_order_book_diffs(self, ev_loop: asyncio.BaseEventLoop, outp
     async def listen_for_order_book_snapshots(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
         while True:
             try:
-                async with aiohttp.ClientSession() as client:
-                    for trading_pair in self._trading_pairs:
-                        try:
-                            snapshot: Dict[str, Any] = await self.get_snapshot(client, trading_pair,
-                                                                               domain=self._domain)
-                            snapshot_timestamp: float = time.time()
-                            snapshot_msg: OrderBookMessage = BinanceOrderBook.snapshot_message_from_exchange(
-                                snapshot,
-                                snapshot_timestamp,
-                                metadata={"trading_pair": trading_pair}
-                            )
-                            output.put_nowait(snapshot_msg)
-                            self.logger().debug(f"Saved order book snapshot for {trading_pair}")
-                            # Be careful not to go above Binance's API rate limits.
-                            await asyncio.sleep(5.0)
-                        except asyncio.CancelledError:
-                            raise
-                        except Exception:
-                            self.logger().error("Unexpected error.", exc_info=True)
-                            await asyncio.sleep(5.0)
-                    this_hour: pd.Timestamp = pd.Timestamp.utcnow().replace(minute=0, second=0, microsecond=0)
-                    next_hour: pd.Timestamp = this_hour + pd.Timedelta(hours=1)
-                    delta: float = next_hour.timestamp() - time.time()
-                    await asyncio.sleep(delta)
+                for trading_pair in self._trading_pairs:
+                    try:
+                        snapshot: Dict[str, Any] = await self.get_snapshot(trading_pair, domain=self._domain)
```

Comment:
In this case, we're able to pass the throttler of the instance to `get_snapshot`